action.yml

```diff
@@ -13,7 +13,16 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - name: Checkout wasp-app-runner
+    - name: Checkout wasp-app-runner (current repo, use triggering commit)
+      if: ${{ github.repository == 'wasp-lang/runner-action' }}
+      uses: actions/checkout@v4
+      with:
+        repository: wasp-lang/runner-action
+        path: ${{ inputs['checkout-path'] }}
+        ref: ${{ github.sha }}
+
+    - name: Checkout wasp-app-runner (external repo, use default branch)
+      if: ${{ github.repository != 'wasp-lang/runner-action' }}
```

Comment:
While this is a smart solution, I think it can be made simpler:

- Add an input to this action like `app-runner-ref` or whatever. Not required, no default value.
- Have this step only once, passing it the new input as the `ref`
- In the `test-action` workflow, manually add `app-runner-ref: ${{ github.sha }}`

This way we don't have a split in the action, but we will still be using the triggering commit when we need it